### DISCUSSION TITLE
Spawning tasks in Windows

### DISCUSSION THREAD

**Discussion by dimi309** [zed-industries/zed#12575] [Feature Requests]
I find tasks to be a very useful feature since it allows me for example to launch a `cargo run` command on a project I am working on from within zed. 

Unfortunately one such task I have created does not work in Windows while it works fine on MacOS. The problem seems to be the arguments passed to the shell command. It looks to me like they are only suitable for MacOS and maybe Linux systems. I have added a condition there to change these arguments (using -C insted of -i -c) if the SHELL envrionment variable contains "powershell". It works for my local build of Zed. 

I would like to contribute it if it can help advance the Windows edition, but I thought to start this discussion before trying to introduce a pull request, since I do not know if the team is interested in such improvements at the moment, or if they are already planning to handle this issue in a way that is more generic than my two if/then snippets. 😄 Any feedback would be greatly appreciated.

![image](https://github.com/zed-industries/zed/assets/57290206/8b0a89e7-c6b2-463b-8bcc-de6af6d57cab)

**Comment by GitYun** [comment]
I use newest build of Zed in Windows11, the problem hasn't been fixed yet.

**Comment by ArhumMK** [comment]
I use Zed Dev 0.160.0 and no matter what i define in tasks.json it runs everything with "powershell -C" thus none of the tasks work, not even the example task. This would be better as an issue i think.

**Reply by dimi309** [reply]
I would be happy to describe the problem in an issue and provide my solution in a pull request (adapted to feedback if some is provided by the Zed team). In the meantime, I have started this discussion since that is what one should do as a first step, according to the documentation (https://github.com/zed-industries/zed/blob/main/CONTRIBUTING.md)

**Comment by dimi309** [comment]
Actually I have just cloned the latest version of the source code and built it locally. It would appear that they have already fixed the issue!!! 😃 If powershell does not work in the current built release I am sure it will work soon, in an upcoming one.

![image](https://github.com/user-attachments/assets/e7ca7b11-1af4-40de-aa93-418de0a9c219)
